runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/PubsubIOTranslator.java

```diff
@@ -32,32 +34,25 @@
   /**
    * Implements PubsubIO Read translation for the Dataflow backend.
    */
-  public static class ReadTranslator<T> implements TransformTranslator<PubsubIO.Read.Bound<T>> {
+  public static class ReadTranslator<T> implements TransformTranslator<PubsubUnboundedSource<T>> {
     @Override
     @SuppressWarnings({"rawtypes", "unchecked"})
     public void translate(
-        PubsubIO.Read.Bound transform,
+        PubsubUnboundedSource<T> transform,
         TranslationContext context) {
       translateReadHelper(transform, context);
     }
 
     private <T> void translateReadHelper(
-        PubsubIO.Read.Bound<T> transform,
+        PubsubUnboundedSource<T> transform,
         TranslationContext context) {
-      if (!context.getPipelineOptions().isStreaming()) {
-        throw new IllegalArgumentException(
-            "PubsubIO.Read can only be used with the Dataflow streaming runner.");
-      }
+      Preconditions.checkState(context.getPipelineOptions().isStreaming(),
+          "PubsubIOTranslator.ReadTranslator is only for streaming pipelines.");
 
       context.addStep(transform, "ParallelRead");
       context.addInput(PropertyNames.FORMAT, "pubsub");
-      if (transform.getTopic() != null) {
-        context.addInput(PropertyNames.PUBSUB_TOPIC, transform.getTopic().asV1Beta1Path());
```

Comment:
I had to give up supporting the 'create random subscription if only given a topic' since there's no way to know when to clean it up.